### PR TITLE
Poetry

### PR DESCRIPTION
**Pull request by vtnate**
#### Any background context you want to provide?
Our setup process was getting too complicated. Using Poetry lets us create a single venv that everyone uses, while not being a Docker container.
Commit message for 4f84004 has a typo. The word should be `Poetry`, not `python`.
#### What does this PR accomplish?
Removes `setup.py` and `requirements.txt` in favor of Poetry's `pyproject.toml` config file.

Things that need work before this is ready:
- [x] The test command below fails the `teaser_single` test, though it may be a flake8 error, I'm not quite sure what the error is telling me.
- [x] I nuked the docs test in `tox.ini` because it contains a reference to `setup.py` which no longer exists with Poetry. Need help to bring that back.
- [x] We no longer have the `package` and `cmdclass` features from the old `setup.py`. Not sure if we need those or not.
- [x] I updated the `requests` package to 2.25.1, after checking that nothing appears to be removed between [that update](https://requests.readthedocs.io/en/latest/community/updates/#release-history) and 2.24.0, our previous version. Double check that this is ok?
- [x] The pre-commit hook for formatting Modelica files doesn't work. I'd love some help sorting that out.

I haven't found any way for the venv to activate/deactivate automatically as you move in/out of the project dir, a la [pyenv-virtualenv](https://github.com/pyenv/pyenv-virtualenv#activate-virtualenv). Would be cool if that was a thing.

#### How should this be manually tested?
`poetry install`
`poetry run tox`
#### What are the relevant tickets?
#299

### CODE REVIEW

**Review by macintoshpie — COMMENTED** [review]
> We no longer have the package and cmdclass features from the old setup.py. Not sure if we need those or not.

I know format_modelica_files is used, but not sure about the outhers. I think cmdclasses will need to be refactored into functions if possible and registered as scripts: https://python-poetry.org/docs/pyproject/#scripts

> I updated the requests package to 2.25.1, after checking that nothing appears to be removed between that update and 2.24.0, our previous version. Double check that this is ok?

If the tests pass we should be good. Looks like we're only explicitly using requests

> The pre-commit hook for formatting Modelica files doesn't work. I'd love some help sorting that out

I think that probably has to do with the comment above, it needs to be a function 

> I haven't found any way for the venv to activate/deactivate automatically as you move in/out of the project dir, a la pyenv-virtualenv. Would be cool if that was a thing.

I've never done something like this, but if you wanted to get fancy you could maybe use something like [chpwd](https://www.refining-linux.org/archives/42-ZSH-Gem-8-Hook-function-chpwd.html), detect if directory is a poetry project, then run `poetry shell`.

**Comment by vtnate** [comment]
@macintoshpie I made some more changes that still don't work, specifically the `precommit` and `docs` tests from `tox.ini`.

I'm also not sure if the `[tool.poetry.scripts]` for schema & license are running. I did confirm the cli works, before my first push.

Can you help me some more with this?

File: tox.ini [hunk 1]
```diff
@@ -1,36 +1,41 @@
...
     requests
-commands=pre-commit run --all-files
+commands=
+    poetry install --no-root -v
+    poetry run pre-commit run -a
 whitelist_externals=
     modelicafmt
+    poetry
 
 [testenv:docs]
```
Comment (macintoshpie) [inline tox.ini:32]: Try this instead:
```
[testenv:docs]
changedir=docs
commands=
    poetry install --no-root -v
    poetry run make html
whitelist_externals=
    poetry
    make
```
I found this issue which lead me to this: https://github.com/tox-dev/tox/issues/363

**Comment by vtnate** [comment]
@nllong Are these 3 `expected` checks a remnant from my changes to [the ci config](https://github.com/urbanopt/geojson-modelica-translator/blob/poetry/.github/workflows/ci.yml#L14)? The checks don't ever run, is that because I did something strange setting up the ci environments?

**Comment by nllong** [comment]
> @nllong Are these 3 `expected` checks a remnant from my changes to [the ci config](https://github.com/urbanopt/geojson-modelica-translator/blob/poetry/.github/workflows/ci.yml#L14)? The checks don't ever run, is that because I did something strange setting up the ci environments?

I'm unsure why there are 3 extra checks. I'll review the PR and maybe if there is a change needed it will force a rerun. ┐( ∵ )┌

**Comment by macintoshpie** [comment]
@nllong @vtnate yeah, now that use use a different test matrix and removed the `formatting` job it is different. will need to update the repo settings

**Review by nllong — APPROVED** [review]
Looks awesome. I tested this locally and found a few things. Can you review this commit to make sure all is good?

https://github.com/urbanopt/geojson-modelica-translator/pull/297/commits/a2d8158a1418a9a0ec6d1b9d7eadf75d8c61b7d2

File: geojson_modelica_translator/model_connectors/load_connectors/spawn.py [hunk 2]
```diff
@@ -21,9 +21,7 @@
 IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
 FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
 CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
-DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUR
-
-EMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
```
[the remark this answers is not in the document]
Comment (vtnate) [reply geojson_modelica_translator/model_connectors/load_connectors/spawn.py:24]: During testing I ran the update_licenses script with poetry, before you changed it to include 2021. The script works! 😁

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -9,8 +9,9 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        # 3.8, 3.9 do not work at the moment.
+        # 3.9 does not work as of 2020-02-01
```
[the remark this answers is not in the document]
Comment (vtnate) [reply .github/workflows/ci.yml:12]: In my testing Pandas still has some hiccups with 3.9, I think we should wait to explicitly support 3.9.

**Comment by vtnate** [comment]
@nllong I made a small change to suggest using tox instead of pytest in the readme. Otherwise 👍. Still need to make those 3 hanging tests go away: can you look at the [updated ci file](https://github.com/urbanopt/geojson-modelica-translator/blob/poetry/.github/workflows/ci.yml) for what could be causing those?

File: pyproject.toml [hunk 4]
```diff
@@ -50,5 +50,5 @@ build-backend = "poetry.masonry.api"
 [tool.poetry.scripts]
 uo_des = 'management.uo_des:cli'
 format_modelica_files = 'management.format_modelica_files:fmt_modelica_files'
-update_schemas = 'management.update_schemas:update_licenses'
+update_schemas = 'management.update_schemas:update_schemas'
```
Comment (vtnate) [inline pyproject.toml:53]: 😅

**Comment by nllong** [comment]
> @nllong I made a small change to suggest using tox instead of pytest in the readme. Otherwise 👍. Still need to make those 3 hanging tests go away: can you look at the [updated ci file](https://github.com/urbanopt/geojson-modelica-translator/blob/poetry/.github/workflows/ci.yml) for what could be causing those?

Cool, just updated the test requirements!

**Pull request merged**